components/org.wso2.carbon.deployment.engine/src/main/java/org/wso2/carbon/deployment/engine/internal/DeploymentEngine.java

```diff
@@ -93,15 +98,22 @@ public DeploymentEngine() {
      * This will start the repository scanner and scheduler task and load artifacts to
      * the deployment engine.
      *
-     * @param repositoryDir the deployment repository directory that repository scanner will start scanning.
+     * @param serverRepositoryDir server deployment repository directory that repository scanner will start scanning.
+     * @param runtimeRepositoryDir runtime deployment repository directory that repository scanner will start scanning.
      * @throws DeploymentEngineException when an error occurs while trying to start the deployment engine.
      */
-    public void start(String repositoryDir) throws DeploymentEngineException {
-        logger.debug("Starting carbon deployment engine for repository : " + repositoryDir);
-        repositoryDirectory = new File(repositoryDir);
-        if (!repositoryDirectory.exists()) {
-            throw new DeploymentEngineException("Cannot find repository : " + repositoryDirectory);
+    public void start(String serverRepositoryDir, String runtimeRepositoryDir) throws DeploymentEngineException {
+        serverRepositoryDirectory = new File(serverRepositoryDir);
+        runtimeRepositoryDirectory = new File(runtimeRepositoryDir);
+
+        if (!serverRepositoryDirectory.exists()) {
+            throw new DeploymentEngineException("Cannot find repository : " + serverRepositoryDirectory);
+        }
+        if (!runtimeRepositoryDirectory.exists()) {
+            throw new DeploymentEngineException("Cannot find repository : " + runtimeRepositoryDirectory);
         }
+        logger.debug("Starting carbon deployment engine for repository : " + serverRepositoryDir);
```

Comment:
Those are the server deployment and runtime specific deployment dirs where Deployers will scan periodically for artifacts and deploy